pkg/controller/endpointslice/topologycache/topologycache.go

```diff
@@ -88,12 +88,12 @@ func (t *TopologyCache) GetOverloadedServices() []string {
 
 // AddHints adds or updates topology hints on EndpointSlices and returns updated
 // lists of EndpointSlices to create and update.
-func (t *TopologyCache) AddHints(si *SliceInfo) ([]*discovery.EndpointSlice, []*discovery.EndpointSlice, []*EventBuilder) {
+func (t *TopologyCache) AddHints(logger klog.Logger, si *SliceInfo) ([]*discovery.EndpointSlice, []*discovery.EndpointSlice, []*EventBuilder) {
 	totalEndpoints := si.getTotalReadyEndpoints()
 	allocations, allocationsEvent := t.getAllocations(totalEndpoints)
 	events := []*EventBuilder{}
 	if allocationsEvent != nil {
-		klog.InfoS(allocationsEvent.Message+", removing hints", "serviceKey", si.ServiceKey, "addressType", si.AddressType)
+		logger.Info(allocationsEvent.Message+", removing hints", "serviceKey", si.ServiceKey, "addressType", si.AddressType)
```

Comment:
Is this the same "key" as above?

Reply:
if my understanding of code is correct, they are same. but I'm not fully sure. Can you please help confirming it?

Reply:
The work queue key is `namespace/name`, just as `si.ServiceKey` (https://github.com/kubernetes/kubernetes/blob/ddbf3575a7f32503c02c1894a1d7dff8c29e766a/pkg/controller/endpointslice/reconciler.go#L257).

It's just that conceptually they are different (= different pieces of the code). Tricky.

I think we should use "key" instead of "serviceKey". It's shorter and if someone searches for "key=xxx" it's better to get more log output (even if it happens to be unrelated) than not getting it (which would happen if this code used "serviceKey").

Reply:
I agree 👍